### PR TITLE
Consistent support for DELIVER_USER and DELIVER_USERNAME env variables

### LINKED ISSUE #6478: Inconsistent ENV Variables used
### New Issue Checklist
- [X] Updated fastlane to the latest version
- [X] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
### Issue Description

Inconsistent environment variables are used throughout fastlane leading to duplicate entries for tools.

action `latest_testflight_build_number` expects ITUNESCONNECT_USER
Sigh expects SIGH_USERNAME
Deliver expects DELIVER_USERNAME
action `set_changelog` expect FASTLANE_USERNAME
Credentials Manager expects FASTLANE_USER
Produce expects DELIVER_USER or PRODUCE_USERNAME
watchbuild expects FASTLANE_USER
etc...

**Suggestion 1**
Enforce standard naming of environment variables for per-tool settings

```
{tool_name}_USERNAME
{tool_name}_PASSWORD
```

which override any global settings, for example

```
ITUNESCONNECT_USERNAME
ITUNESCONNECT_PASSWORD (or use credential manager)
MEMBERCENTER_USERNAME
MEMBERCENTER_PASSWORD (or use credential manager)
```

or for using the same credentials for all Apple/iTunes related work

```
FASTLANE_USERNAME
FASTLANE_PASSWORD (or use credential manager)
```

**Suggestion 2**
Create documentation listing all global and tool-specific environment variables.

I welcome some feedback and discussion on this.

### PR DESCRIPTION
**Pull request by asfalcone**
Fixes #6478

Replaces https://github.com/fastlane/fastlane/pull/6527. I made a new PR for timeliness because the previous one had merge conflicts and it was 3 weeks since the last comment.

### CODE REVIEW

File: deliver/lib/deliver/options.rb [hunk 1]
```diff
@@ -6,6 +6,7 @@ class Options
     def self.available_options
       user = CredentialsManager::AppfileConfig.try_fetch_value(:itunes_connect_id)
       user ||= CredentialsManager::AppfileConfig.try_fetch_value(:apple_id)
+      user ||= ENV["DELIVER_USER"]
```
Comment (KrauseFx) [inline deliver/lib/deliver/options.rb]: We already support `env_name: "DELIVER_USERNAME"` 3 lines below, seems like it's ben inconsistent so far

**Review by KrauseFx — APPROVED** [review]

**Comment by KrauseFx** [comment]
Are there other spots we need to update this for?

File: deliver/lib/deliver/options.rb (hunk 1 above)
Comment (asfalcone) [inline deliver/lib/deliver/options.rb]: DELIVER_USER is referenced in [credentials manager.](https://github.com/fastlane/fastlane/blob/master/credentials_manager/lib/credentials_manager/appfile_config.rb#L76) Maybe we need to add `DELIVER_USERNAME` to there as well?

Comment (asfalcone) [inline deliver/lib/deliver/options.rb]: Added it there @KrauseFx

**Review by KrauseFx — APPROVED** [review]

**Pull request merged**